x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -4,383 +4,74 @@
-      throw new Error(
-        `Rule with id ${ruleId} not found. Could not fetch linked dashboards for alert with id ${this.alertId}.`
-      );
-    }
-    const linkedDashboardsArtifacts = rule.artifacts?.dashboards || [];
-    const linkedDashboards = await this.getLinkedDashboardsByIds(
-      linkedDashboardsArtifacts.map((d) => d.id)
-    );
-    return linkedDashboards;
-  }
-
-  private async getLinkedDashboardsByIds(ids: string[]): Promise<LinkedDashboard[]> {
-    const linkedDashboardsResponse = await Promise.all(
-      ids.map((id) => this.getLinkedDashboardById(id))
-    );
-    return linkedDashboardsResponse.filter((dashboard): dashboard is LinkedDashboard =>
-      Boolean(dashboard)
-    );
-  }
-
-  private async getLinkedDashboardById(id: string): Promise<LinkedDashboard | null> {
-    try {
-      const dashboardResponse = await this.dashboardClient.get(id);
-      return {
-        id: dashboardResponse.result.item.id,
-        title: dashboardResponse.result.item.attributes.title,
-        matchedBy: { linked: true },
-        description: dashboardResponse.result.item.attributes.description,
-        tags: dashboardResponse.result.item.attributes.tags,
-      };
-    } catch (error) {
-      if (error.output.statusCode === 404) {
-        this.logger.warn(`Linked dashboard with id ${id} not found. Skipping.`);
-        return null;
-      }
-      throw new Error(`Error fetching dashboard with id ${id}: ${error.message || error}`);
-    }
-  }
+    const linkedDashboardIds = (
+      (await await this.alertsClient.getRuleById(ruleId))?.artifacts?.dashboards || []
+    ).map((d) => d.id);
 
-  private getMatchingFields(dashboard: RelatedDashboard): string[] {
-    const matchingFields = new Set<string>();
-    // grab all the top level arrays from the matchedBy object via Object.values
-    Object.values(omit(dashboard.matchedBy, 'linked')).forEach((match) => {
-      // add the values of each array to the matchingFields set
-      match.forEach((value) => {
-        matchingFields.add(value);
-      });
-    });
-    return Array.from(matchingFields);
-  }
+    const [suggestedDashboards, linkedDashboards] = await Promise.all([
+      this.suggestedDashboardsClient.fetchSuggested(alert),
+      this.linkedDashboardsClient.getLinkedDashboardsByIds(linkedDashboardIds),
+    ]);
 
-  private getScore(dashboard: RelatedDashboard): number {
-    const alert = this.checkAlert();
-    const allRelevantFields = alert.getAllRelevantFields();
-    const index = this.getRuleQueryIndex();
-    const setA = new Set<string>([...allRelevantFields, ...(index ? [index] : [])]);
-    const setB = new Set<string>(this.getMatchingFields(dashboard));
-    const intersection = new Set([...setA].filter((item) => setB.has(item)));
-    const union = new Set([...setA, ...setB]);
+    const filteredSuggestedDashboards = suggestedDashboards.filter(
+      (suggested) => !linkedDashboards.some((linked) => linked.id === suggested.id)
+    );
```

Comment:
```suggestion
    const filteredSuggestedDashboards = suggestedDashboards.filter(
      (suggested) => !linkedDashboards.some((linked) => linked.id === suggested.id)
    ).slice(0, 10);
```

Recommend moving the slice here instead of computing at the end of the function. This is purely cosmetic but keeps the return cleaner.

If possible it would be nice if we could limit the result size at query time, but if there are technical constraints related to filtering and ranking that prevent this that makes sense.